diff --git a/util/mkinstallvars.pl b/util/mkinstallvars.pl
--- a/util/mkinstallvars.pl
+++ b/util/mkinstallvars.pl
@@ -35,7 +35,6 @@
 foreach (@absolutes) { $all{$_} = 1 }
 foreach (@subdirs) { foreach (@{$_->[1]}) { $all{$_} = 1 } }
 foreach (@others) { $all{$_} = 1 }
-print STDERR "DEBUG: all keys: ", join(", ", sort keys %all), "\n";
 
 my %keys = ();
 my %values = ();
@@ -45,6 +44,13 @@
     push @{$values{$k}}, $v;
 }
 
+foreach my $optional (qw(COMMENT CMAKECONFIGDIR PKGCONFIGDIR libdir)) {
+    next if $keys{$optional};
+    $keys{$optional} = 1;
+    $values{$optional} =
+        $optional eq 'COMMENT' || $optional eq 'libdir' ? [ '' ] : [ '.' ];
+}
+
 # special case for LIBDIR vs libdir.
 # For installations, They both get their value from ./Configure's --libdir or
 # corresponding config target attribute, but LIBDIR only gets a value if the
@@ -52,9 +58,7 @@
 # the former doesn't have a value, we give it the latter's value, and rely
 # on mechanisms further down to do the rest of the processing.
 # If they're both empty, it's still fine.
-print STDERR "DEBUG: LIBDIR = $values{LIBDIR}->[0], libdir = $values{libdir}->[0] => ";
 $values{LIBDIR}->[0] = $values{libdir}->[0] unless $values{LIBDIR}->[0];
-print STDERR "LIBDIR = $values{LIBDIR}->[0]\n";
 
 # warn if there are missing values, and also if there are unexpected values
 foreach my $k (sort keys %all) {
@@ -69,10 +73,8 @@
 foreach my $k (@absolutes) {
     my $v = $values{$k} || [ '.' ];
     die "Can't have more than one $k\n" if scalar @$v > 1;
-    print STDERR "DEBUG: $k = $v->[0] => ";
     $v = [ map { File::Spec->rel2abs($_) } @$v ];
     $values{$k} = $v;
-    print STDERR "$k = $v->[0]\n";
 }
 
 # Absolute paths for the subdir variables are computed.  This provides
@@ -88,9 +90,6 @@
         my $kr = "${k}_REL_${var}";
         my $v2 = $values{$k} || [ '.' ];
         $values{$k} = [];       # We're rebuilding it
-        print STDERR "DEBUG: $k = ",
-            (scalar @$v2 > 1 ? "[ " . join(", ", @$v2) . " ]" : $v2->[0]),
-            " => ";
         foreach my $v (@$v2) {
             if (File::Spec->file_name_is_absolute($v)) {
                 push @{$values{$k}}, $v;
@@ -102,14 +101,6 @@
                     File::Spec->rel2abs($v, $values{$var}->[0]);
             }
         }
-        print STDERR join(", ",
-                          map {
-                              my $v = $values{$_};
-                              "$_ = " . (scalar @$v > 1
-                                         ? "[ " . join(", ", @$v) . " ]"
-                                         : $v->[0]);
-                          } ($k, $kr)),
-            "\n";
     }
 }
 
